### PR TITLE
Students get assigned to account as dependent or primary account holder

### PR DESCRIPTION
**Pull request by cassie1992**
when created. Students have DOB field. Added validation tests and some
other tests

### CODE REVIEW

File: .classpath [hunk 1]
```diff
@@ -1,12 +1,12 @@
...
+        <classpathentry kind="src" path="src"/>
+        <classpathentry kind="con" path="com.google.appengine.eclipse.core.GAE_CONTAINER"/>
+        <classpathentry kind="con" path="org.eclipse.jdt.launching.JRE_CONTAINER"/>
+        <classpathentry kind="con" path="org.eclipse.jdt.junit.JUNIT_CONTAINER/4"/>
+        <classpathentry kind="var" path="ECLIPSE_HOME/plugins/com.google.appengine.eclipse.sdkbundle_1.8.8/appengine-java-sdk-1.8.8/lib/impl/appengine-api.jar"/>
+        <classpathentry kind="var" path="ECLIPSE_HOME/plugins/com.google.appengine.eclipse.sdkbundle_1.8.8/appengine-java-sdk-1.8.8/lib/impl/appengine-api-labs.jar"/>
+        <classpathentry kind="var" path="ECLIPSE_HOME/plugins/com.google.appengine.eclipse.sdkbundle_1.8.8/appengine-java-sdk-1.8.8/lib/impl/appengine-api-stubs.jar"/>
+        <classpathentry kind="var" path="ECLIPSE_HOME/plugins/com.google.appengine.eclipse.sdkbundle_1.8.8/appengine-java-sdk-1.8.8/lib/testing/appengine-testing.jar"/>
+        <classpathentry kind="output" path="war/WEB-INF/classes"/>
+</classpath>
```
Comment (Ajedi32) [inline .classpath:12]: I noticed you did try to revert this, but there are still indentation changes here.

File: src/META-INF/jdoconfig.xml [hunk 2]
```diff
@@ -1,16 +1,13 @@
...
+   <persistence-manager-factory name="transactions-optional">
+       <property name="javax.jdo.PersistenceManagerFactoryClass" value="org.datanucleus.api.jdo.JDOPersistenceManagerFactory"/>
+       <property name="javax.jdo.option.ConnectionURL" value="appengine"/>
+       <property name="javax.jdo.option.NontransactionalRead" value="true"/>
+       <property name="javax.jdo.option.NontransactionalWrite" value="true"/>
+       <property name="javax.jdo.option.RetainValues" value="true"/>
+       <property name="datanucleus.appengine.autoCreateDatastoreTxns" value="true"/>
+       <property name="datanucleus.appengine.singletonPMFForName" value="true"/>
+   </persistence-manager-factory>
+</jdoconfig>
```
Comment (Ajedi32) [inline src/META-INF/jdoconfig.xml:13]: Any idea what these changes are for? If not, I'll probably just revert it.

File: src/META-INF/persistence.xml [hunk 3]
```diff
@@ -1,15 +1,12 @@
...
+
+    <persistence-unit name="transactions-optional">
+        <provider>org.datanucleus.api.jpa.PersistenceProviderImpl</provider>
+        <properties>
+            <property name="datanucleus.NontransactionalRead" value="true"/>
+            <property name="datanucleus.NontransactionalWrite" value="true"/>
+            <property name="datanucleus.ConnectionURL" value="appengine"/>
+        </properties>
+    </persistence-unit>
+</persistence>
```
Comment (Ajedi32) [inline src/META-INF/persistence.xml:12]: Ditto for this file.

File: src/edu/uwm/cs361/fantastic_five/training_tracker/app/entities/Program.java [hunk 4]
```diff
@@ -27,7 +27,6 @@
 	@Persistent
 	private double price;
 	
-	
 	@Persistent
```
Comment (Ajedi32) [inline src/edu/uwm/cs361/fantastic_five/training_tracker/app/entities/Program.java:31]: Random whitespace change?

File: src/edu/uwm/cs361/fantastic_five/training_tracker/app/entities/Student.java [hunk 5]
```diff
@@ -105,17 +109,9 @@ public void setEmail(String e) {
...
-	}
 
-	public String balanceToString(){
-		return String.format("$%.2f", new Double(balance));
-	}
-
-	public void updateBalance(double price){
-		balance+=price;
+	public String getDOB() {
+		return DOB;
```
Comment (Ajedi32) [inline src/edu/uwm/cs361/fantastic_five/training_tracker/app/entities/Student.java:114]: Lots of seemingly unrelated changes in this pull request. If possible, try to submit unrelated changes in seperate pull requests if you can.

**Comment by cassie1992** [comment]
Yeah sorry about all the extra changes, I meant to only commit certain files but I think it automatically committed everything at once when I hit commit

**Comment by Ajedi32** [comment]
Yeah, I'm not sure what GUI you're using but there should be some option when committing that lets you select which files to commit.

File: src/META-INF/jdoconfig.xml (hunk 2 above)
Comment (cassie1992) [inline src/META-INF/jdoconfig.xml:13]: I'm pretty sure these files got changed by eclipse so I have no idea. It looks like it just changed the spacing

**Comment by cassie1992** [comment]
Added a new commit. I tried not to change the classpath but it did anyways, so I fixed it again

**Comment by Ajedi32** [comment]
Okay, looks like there's still a whitespace change in the classpath but don't worry about that for now; I'll fix it when I merge.

File: src/edu/uwm/cs361/fantastic_five/training_tracker/app/use_cases/AccountCreator.java [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+				resp.success = false;
+			}
+		
+		return resp;
+	}
+	private PersistenceManager getPersistenceManager()
+	{
+		return PersistenceService.getPersistenceManager();
+	}
+}
```
Comment (Ajedi32) [inline src/edu/uwm/cs361/fantastic_five/training_tracker/app/use_cases/AccountCreator.java:50]: No tests for this...

File: src/edu/uwm/cs361/fantastic_five/training_tracker/app/use_cases/AccountFinder.java [hunk 7]
```diff
@@ -0,0 +1,32 @@
...
+
+	public ListAccountsResponse listAccounts() {
+		return listAccounts(new ListAccountsRequest());
+	}
+
+	private PersistenceManager getPersistenceManager()
+	{
+		return PersistenceService.getPersistenceManager();
+	}
+}
```
Comment (Ajedi32) [inline src/edu/uwm/cs361/fantastic_five/training_tracker/app/use_cases/AccountFinder.java:32]: No tests for this either as far as I can tell...

File: src/edu/uwm/cs361/fantastic_five/training_tracker/servlets/AccountFormServlet.java [hunk 8]
```diff
@@ -0,0 +1,69 @@
...
+			CreateAccountResponse createResponse = new AccountCreator().createAccount(createRequest);
+	
+			if (createResponse.success) {
+				resp.sendRedirect("/students");
+			} else {
+				req.setAttribute("errors", createResponse.errors);
+				forwardToJsp("create_account_form.jsp",req,resp);
+			}
+		}
+	}
```
Comment (Ajedi32) [inline src/edu/uwm/cs361/fantastic_five/training_tracker/servlets/AccountFormServlet.java:68]: Looks like there's a lot of redundancy here. This could probably be much more [DRY](http://en.wikipedia.org/wiki/Don't_repeat_yourself).

**Comment by Ajedi32** [comment]
Merged into the integration branch. If you want to address any of the comments I've made here, feel free to submit another pull request with the appropriate fixes.

**Comment by Ajedi32** [comment]
FYI, this merge broke the build. A whole bunch of tests are failing now.

The primary cause seems to be that `CreateProgramRequest->instructor` is now expected to be an ID and not a string.

All the tests for `StudentCreator#createStudent` are also failing due to a null pointer exception in the `StudentValidator` class, which seems to be caused by the tests not adding a date of birth to the request object. Care to try fixing these errors?

**Comment by cassie1992** [comment]
I can fix those, I'm also working on the other tests. I'll have it up by the morning

**Comment by Ajedi32** [comment]
Okay, great. :+1:  Send me a pull request when those fixes are ready.
